app/selectors/i18n.js

```diff
@@ -11,7 +11,7 @@ import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
 export const getCurrentUserLocale = createSelector(
     getCurrentUser,
     (currentUser) => {
-        return currentUser ? currentUser.locale : '';
+        return currentUser ? currentUser.locale : 'en';
```

Comment:
This should be returning the empty string so that getCurrentLocale picks up the device locale properly. 

You should be using getCurentLocale instead of this function, and we should stop exporting this function since it's clearly confusing :)